frontend/src/app/components/wp-query-select/wp-query-select-dropdown.component.ts

```diff
@@ -170,10 +170,12 @@ export class WorkPackageQuerySelectDropdownComponent implements OnInit, OnDestro
     this.updateMenuOnChanges(this.searchInput);
   }
 
-  private transformQueries(collection:CollectionResource) {
-    let loadedQueries:IAutocompleteItem[] = _.map(collection.elements, (query:any) => {
-      return {label: query.name, query: query, query_props: null};
-    });
+  private transformQueries(collection:CollectionResource<QueryResource>) {
+    let loadedQueries:IAutocompleteItem[] = collection.elements
+      .filter(query => !query.hidden)
+      .map(query => {
+        return {label: query.name, query: query, query_props: null};
```

Comment:
It would be more efficient to implement a 'hidden' filter in the backend which is used here. Hidden queries would then not have to be fetched from the db, rendered and finally transmitted to the front end in the first place.

Reply:
Good point, done!